File: 58-length-of-last-word/length-of-last-word.js
/**
 * @param {string} s
 * @return {number}
 */
function lengthOfLastWord(s) {
    // Trim the string to remove trailing spaces
    s = s.trim();
    
    // Split the string into words
    const words = s.split(' ');
    
    // Return the length of the last word
    return words[words.length - 1].length;
}

File: 58-length-of-last-word/length-of-last-word.cpp
#include <string>
#include <sstream>
using namespace std;

class Solution {
public:
    int lengthOfLastWord(string s) {
        // Trim trailing spaces
        int n = s.length();
        int length = 0;

        // Start from the end of the string and count non-space characters
        for (int i = n - 1; i >= 0; --i) {
            if (s[i] == ' ') {
                if (length > 0) {
                    break; // Stop counting when we hit the first space after counting the last word
                }
            } else {
                length++; // Count the length of the last word
            }
        }
        
        return length;
    }
};